Update checklist and CONTRIBUTING.md based on automation changes (#274)
AlexsLemonade/refinebio-examples: CONTRIBUTING.md

## Rendering notebooks

> ### Automatic rendering using GitHub actions

Reviewer: Since this happens _after_ the PR has been accepted, we should probably move this section to after the part where we explain local rendering. This is a nice thing to know about, but if it is working properly, they should not run into it at all.

Author: See if you agree with where I moved it to.

This repository uses [snakemake](https://snakemake.readthedocs.io/en/stable/getting_started/installation.html) to render all notebooks.

> All notebooks are automatically re-rendered by GitHub actions before merges to master.

Reviewer: This happens after not before...
```suggestion
All notebooks are automatically re-rendered by GitHub actions upon merges to master.
```

The newly rendered html files are all pushed to the `gh-pages` branch which will publish the material to https://alexslemonade.github.io/refinebio-examples/.

> If this automatic rendering fails, you can see the errors on GitHub by going to `Actions` and clicking the workflow of PR you are working on that also says `Build Docker` underneath.

Reviewer: Two things here:
1. How will they know it has failed? (I think they should get an email, but have we tested this? Do we get an email when merge automations fail?)
2. What do they do with a failure? I would encourage them to make changes immediately to fix the error, and or contact maintainers for support.

Author: I get emails when things I push failed, but depending on peoples' notification settings I could see that this isn't necessarily a given? Anyway, I've adjusted this wording, see what you think,

Click on `build` on the left bar and click on the step that has failed to see the error message.

### Mechanics of the rendering

The `Snakefile` calls the `scripts/render-notebooks.R` which renders the `.html` files but leaves these `.Rmd` files ready for download and use [without the `pandoc` error](https://github.com/AlexsLemonade/refinebio-examples/pull/148#issuecomment-669170681).

> However, `snakemake` can (and should) be run locally during development so that the author and reviewers can see the rendered output of the new material during the `Pull Request` process.

Reviewer: I think we should keep the emphasis on "should"
```suggestion
However, the `snakemake` workflow should also be run locally during development so that the author and reviewers can see the rendered output of the new material during the `Pull Request` process.
```

> ### How to re-render the notebooks locally

Reviewer: This change will break the link on line 95: 
`- In the Docker container, run [snakemake for rendering](#how-to-re-render-the-notebooks)`

Author: Good call.

## Rendering notebooks

### Mechanics of the rendering

The `Snakefile` calls the `scripts/render-notebooks.R` which renders the `.html` files but leaves these `.Rmd` files ready for download and use [without the `pandoc` error](https://github.com/AlexsLemonade/refinebio-examples/pull/148#issuecomment-669170681).
However, the `snakemake` workflow should also be run locally during development so that the author and reviewers can see the rendered output of the new material during the `Pull Request` process.

> Ideally snakemake will not re-render the `.html` for `.Rmd` files you have not edited, but if it does, you should only commit and push the files you have intended to change.

Author: Here's my note about not committing "extra" files.

> All `.html` files will be re-rendered upon merging to master so the files will be updated.
> But, by not committing your local, unintentionally altered files, the `Files changed` page of your PR on GitHub will be more focused. 

Reviewer: I made some suggestions to streamline this sentence.
```suggestion
All `.html` files will be re-rendered upon merging to master, but by not committing files that are only altered incidentally, the `Files changed` page of your PR on GitHub will be more focused, easing the burden on reviewers. 
```
